### PR TITLE
Completed task 12 - Article Topic Query

### PR DESCRIPTION
**Pull request by BlurryQ**

### CODE REVIEW

File: controllers/articles.controller.js [hunk 1]
```diff
@@ -8,18 +8,29 @@ const {
...
-    .then((articles) => {
+  const { sort_by, order, topic } = request.query;
+  const unresolvedPromises = [
+    getAllTopics(),
+    selectArticles(sort_by, order, topic),
+  ];
+
+  Promise.all(unresolvedPromises)
+    .then(([topics, articles]) => {
+      if (topic !== undefined && !topics.includes(topic))
```
Comment (nc-hannah-burks) [inline controllers/articles.controller.js:19]: nice validation

File: __tests__/integrations.test.js [hunk 2]
```diff
@@ -196,12 +164,12 @@ describe("Endpoint testing for NC News", () => {
           expect(body.articles).toBeSortedBy("title");
         });
     });
-    it("400: responds with 'bad request' when sort is invalid or doesn't exist", () => {
+    it("200: returns articles filtered by topic", () => {
```
Comment (nc-hannah-burks) [inline __tests__/integrations.test.js:167]: Another test you might want to add is Status 200, valid topic query, but has no articles responds with an empty array of articles, e.g. ?topic=paper.
